### PR TITLE
[tests] make pipelines tests faster with smaller models

### LINKED ISSUE #4234: [infra] make a tiny "distilroberta-base" to speed up test_pipelines.py and test_examples.py
put it in a user namespace on s3 like `sshleifer/tiny-distilroberta`
doing the same for distilbert-base would probably also help.

### LINKED ISSUE #4366: [pipelines] Failing @slow test for TF Summarization
Command:
```bash
RUN_SLOW=1 pytest tests/test_pipelines.py --durations 0 -s -k tf_defaults
```

Issue caused by trying to use Bart in TF.

### PR DESCRIPTION
**Pull request by sshleifer**
- speedup: 8mins -> 90s on 1 CPU

- besides `test_fill_mask` the removed coverage was not meaningful, only shapes were being checked.
- Would be happy to continue for TF or get help.
- could use these in some examples tests also.

### CODE REVIEW

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -258,40 +266,17 @@ def test_tf_feature_extraction(self):
...
 
     @require_torch
-    def test_fill_mask(self):
+    def test_torch_fill_mask(self):
         mandatory_keys = {"sequence", "score", "token"}
         valid_inputs = [
             "My name is <mask>",
             "The largest city in France is <mask>",
         ]
         invalid_inputs = [None]
```
Comment (sshleifer) [inline tests/test_pipelines.py:275]: removed expected_multi_result for speed.

**Review by thomwolf — APPROVED** [review]
Ok for me but I let @julien-c and @mfuntowicz chime in for the pipelines

**Review by mfuntowicz — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Looks good to me!

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM

**Comment by julien-c** [comment]
please wait before merging, I'd like to take a look

File: tests/test_pipelines.py [hunk 2]
```diff
@@ -55,14 +62,14 @@
...
+        "sshleifer/tiny-distilbert-base-uncased-finetuned-sst-2-english",
+        "sshleifer/tiny-distilbert-base-uncased-finetuned-sst-2-english",
     )
 }
 
 TEXT_GENERATION_FINETUNED_MODELS = {
-    ("gpt2", "gpt2"),
-    ("xlnet-base-cased", "xlnet-base-cased"),
+    # model = tokenizer
+    "sshleifer/tiny-gpt2",
```
Comment (julien-c) [inline tests/test_pipelines.py:72]: Can you take advantage of this PR to also move the other ones to just a model identifier?

Comment (julien-c) [reply tests/test_pipelines.py:72]: Will make things cleaner

File: tests/test_pipelines.py (hunk 1 above)
Comment (julien-c) [reply tests/test_pipelines.py:275]: can we keep those integration tests in `@slow`?

**Review by julien-c — CHANGES_REQUESTED** [review]

**Comment by sshleifer** [comment]
Yes, @julien-c I will keep all the old integration test logic in @slow.

File: tests/test_pipelines.py [hunk 3]
```diff
@@ -389,15 +308,29 @@ def test_text_generation(self):
...
+        for model_name in TEXT_GENERATION_FINETUNED_MODELS:
+            nlp = pipeline(task="text-generation", model=model_name, tokenizer=model_name, framework="tf")
             self._test_mono_column_pipeline(
                 nlp, valid_inputs, invalid_inputs, {},
             )
 
 
 class MultiColumnInputTestCase(unittest.TestCase):
-    def _test_multicolumn_pipeline(self, nlp, valid_inputs: list, invalid_inputs: list, output_keys: Iterable[str]):
+    def _test_multicolumn_pipeline(self, nlp):
```
Comment (julien-c) [inline tests/test_pipelines.py:319]: this name should change if its intent is now more specific due to refactoring

File: tests/test_pipelines.py [hunk 4]
```diff
@@ -320,8 +238,9 @@ def test_tf_fill_mask(self):
                 },
             ],
         ]
-        for tokenizer, model, config in TF_FILL_MASK_FINETUNED_MODELS:
-            nlp = pipeline(task="fill-mask", model=model, config=config, tokenizer=tokenizer, framework="tf", topk=2)
+        for model_name in TF_FILL_MASK_FINETUNED_MODELS:
+            tok_arg = (model_name, {"use_fast": False})
```
Comment (julien-c) [inline tests/test_pipelines.py:242]: I don't think we need the ` {"use_fast": False}` (especially as we don't actually test outputs anymore)

File: tests/test_pipelines.py [hunk 5]
```diff
@@ -206,96 +145,75 @@ def _test_mono_column_pipeline(
...
-                    "sequence": "<s> The largest city in France is Lyon</s>",
-                    "score": 0.21112334728240967,
-                    "token": 12790,
-                },
-            ],
-        ]
-        for tokenizer, model, config in FILL_MASK_FINETUNED_MODELS:
-            nlp = pipeline(task="fill-mask", model=model, config=config, tokenizer=tokenizer, topk=2)
+        for model_name in FILL_MASK_FINETUNED_MODELS:
+            tok_arg = (model_name, {"use_fast": False})
```
Comment (julien-c) [inline tests/test_pipelines.py:208]: I don't think we need the ` {"use_fast": False}` (especially as we don't actually test outputs anymore)

File: tests/test_pipelines.py [hunk 6]
```diff
@@ -7,88 +7,27 @@
...
-}
-
-TF_TEXT_CLASSIF_FINETUNED_MODELS = {
-    (
-        "bert-base-uncased",
-        "distilbert-base-uncased-finetuned-sst-2-english",
-        "distilbert-base-uncased-finetuned-sst-2-english",
-    )
-}
+# TF also tested using this s3dir
```
Comment (julien-c) [inline tests/test_pipelines.py:12]:
```suggestion
```

File: tests/test_pipelines.py (hunk 5 above)
Comment (julien-c) [reply tests/test_pipelines.py:208]: This is not integration testing the actual results anymore, right?

Comment (sshleifer) [reply tests/test_pipelines.py:208]: correct, I intend to have two tests for fill_mask, per your earlier comment.  One that is @slow and tests results, one that is fast and doesn't test results.

**Comment by julien-c** [comment]
(Feel free to merge when this is done)

**Pull request merged**